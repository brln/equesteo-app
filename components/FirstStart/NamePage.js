import React, { PureComponent } from 'react'
import {
  ScrollView,
  Text,
  View
} from 'react-native'

import { brand } from '../../colors'
import { logRender } from '../../helpers'
import Button from '../Button'
import NameForm from './NameForm'


export default class NamePage extends PureComponent {
  constructor () {
    super()
    this.changeFirstName = this.changeFirstName.bind(this)
    this.changeLastName = this.changeLastName.bind(this)
    this.nextPage = this.nextPage.bind(this)
  }

  changeFirstName (value) {
    const updated = this.props.user.set('firstName', value)
    this.props.updateUser(updated)
  }

  changeLastName (value) {
    const updated = this.props.user.set('lastName', value)
    this.props.updateUser(updated)
  }

  nextPage () {
    this.props.commitUpdateUser()
    this.props.nextPage()
  }

  render() {
    logRender('NamePage')
    return (
      <View style={{flex: 1}}>
        <View>
          <Text style={{
            fontSize: 30,
            fontFamily: 'Montserrat-Regular',
            color: 'black',
            textAlign: 'center'
          }}>
            What's your name?
          </Text>
        </View>
        <View style={{flex: 1, alignItems: 'center'}}>
          <NameForm
            changeFirstName={this.changeFirstName}
            changeLastName={this.changeLastName}
            nextPage={this.nextPage}
            inputs={this.props.inputs}
            user={this.props.user}
          />
          <View style={{flex: 1, paddingTop: 20}}>
            <Button
              color={brand}
              text={"Go On"}
              onPress={this.nextPage}
            />
          </View>
        </View>
      </View>
    )
  }
}
