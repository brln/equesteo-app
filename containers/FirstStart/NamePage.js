import React, { PureComponent } from 'react'
import {
  ScrollView,
  Text,
  View
} from 'react-native'
import { connect } from 'react-redux'

import { brand } from '../../colors'
import { logRender } from '../../helpers'
import Button from '../../components/Button'
import NameForm from '../../components/FirstStart/NameForm'
import {
  userUpdated,
} from '../../actions/standard'
import {
  persistUserUpdate
} from "../../actions/functional"
import EqNavigation from "../../services/EqNavigation"
import { PROFILE_PHOTO_PAGE } from "../../screens/consts/firstStart"
import Wrapper from '../../components/FirstStart/Wrapper'


class NamePage extends PureComponent {
  static options() {
    return {
      topBar: {
        background: {
          color: 'transparent',
        },
        elevation: 0,
        backButton: {
          color: brand,
        },
      },
      layout: {
        orientation: ['portrait']
      }
    }
  }

  constructor () {
    super()
    this.changeFirstName = this.changeFirstName.bind(this)
    this.changeLastName = this.changeLastName.bind(this)
    this.nextPage = this.nextPage.bind(this)
    this.updateUser = this.updateUser.bind(this)
    this.inputs = []
  }

  changeFirstName (value) {
    const updated = this.props.user.set('firstName', value)
    this.updateUser(updated)
  }

  changeLastName (value) {
    const updated = this.props.user.set('lastName', value)
    this.updateUser(updated)
  }

  nextPage () {
    this.props.dispatch(persistUserUpdate(this.props.user.get('_id'), [], false))
    EqNavigation.push(this.props.componentId, {
      component: {
        name: PROFILE_PHOTO_PAGE,
      }
    }).catch(() => {})
  }

  updateUser (newUser) {
    this.props.dispatch(userUpdated(newUser))
  }


  render() {
    logRender('NamePage')
    return (
      <Wrapper>
        <View style={{flex: 1}}>
          <View>
            <Text style={{
              fontSize: 30,
              fontFamily: 'Montserrat-Regular',
              color: 'black',
              textAlign: 'center'
            }}>
              What's your name?
            </Text>
          </View>
          <View style={{flex: 1, alignItems: 'center'}}>
            <NameForm
              changeFirstName={this.changeFirstName}
              changeLastName={this.changeLastName}
              nextPage={this.nextPage}
              inputs={this.inputs}
              user={this.props.user}
            />
            <View style={{flex: 1, paddingTop: 20}}>
              <Button
                color={brand}
                text={"Go On"}
                onPress={this.nextPage}
              />
            </View>
          </View>
        </View>
      </Wrapper>
    )
  }
}

function mapStateToProps (state, passedProps) {
  const localState = state.get('localState')
  const pouchState = state.get('pouchRecords')
  return {
    user: pouchState.getIn(['users', localState.get('userID')]),
  }
}

export default connect(mapStateToProps)(NamePage)
