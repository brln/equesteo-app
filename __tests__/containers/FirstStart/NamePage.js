import configureStore from 'redux-mock-store'
import React from "react"
import {fromJS, Map} from "immutable"
import renderer from "react-test-renderer"
import { Provider } from 'react-redux'
import thunk from "redux-thunk"

import NamePage from '../../../containers/FirstStart/NamePage'


const mockStore = configureStore([thunk])

test('render', () => {
  const store = mockStore(fromJS({
    localState: {
      userID: 'userID1'
    },
    pouchRecords: {
      users: {
        userID1: {
          _id: 'userID1'
        }
      }
    },
  }))

  const tree = renderer.create(
    <Provider store={store}>
      <NamePage />
    </Provider>)
  expect(tree.toJSON()).toMatchSnapshot()
})
