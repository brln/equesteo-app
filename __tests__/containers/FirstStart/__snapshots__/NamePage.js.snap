// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<View>
  <View
    style={
      Object {
        "height": 1278,
        "position": "absolute",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "flex": 3,
          "justifyContent": "flex-end",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../img/firstStart.jpg",
          }
        }
        style={
          Object {
            "height": 750,
            "resizeMode": "cover",
            "width": 750,
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flexDirection": "column",
            "paddingTop": 20,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "alignItems": "center",
              "height": 60,
              "width": 60,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "black",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 30,
                "textAlign": "center",
              }
            }
          >
            What's your name?
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "paddingTop": 30,
                "width": 495,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 12,
                }
              }
            >
              First
            </Text>
            <TextInput
              allowFontScaling={true}
              autoFocus={true}
              blurOnSubmit={false}
              keyboardType="email-address"
              maxLength={200}
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              rejectResponderTermination={true}
              style={
                Object {
                  "backgroundColor": "white",
                  "borderColor": "black",
                  "borderWidth": 1,
                  "height": 50,
                  "marginBottom": 20,
                  "minWidth": 220,
                  "paddingLeft": 10,
                }
              }
              underlineColorAndroid="transparent"
            />
            <Text
              style={
                Object {
                  "fontSize": 12,
                }
              }
            >
              Last
            </Text>
            <TextInput
              allowFontScaling={true}
              blurOnSubmit={false}
              keyboardType="email-address"
              maxLength={200}
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              rejectResponderTermination={true}
              style={
                Object {
                  "backgroundColor": "white",
                  "borderColor": "black",
                  "borderWidth": 1,
                  "height": 50,
                  "minWidth": 220,
                  "paddingLeft": 10,
                }
              }
              underlineColorAndroid="transparent"
            />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "paddingTop": 20,
              }
            }
          >
            <View
              accessible={true}
              clickable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "#5bc0eb",
                  "borderColor": undefined,
                  "borderRadius": 10,
                  "borderWidth": null,
                  "marginTop": 10,
                  "opacity": 1,
                  "paddingBottom": 10,
                  "paddingTop": 10,
                  "width": undefined,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "textAlign": "center",
                    },
                    undefined,
                  ]
                }
              >
                Go On
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
